be/src/olap/file_helper.cpp

```diff
@@ -44,9 +45,19 @@ FileHandler::FileHandler() :
         _is_using_cache(false),
         _cache_handle(NULL) {
     static std::once_flag once_flag;
-    std::call_once(once_flag, [] {
-        _s_fd_cache = new_lru_cache(config::file_descriptor_cache_capacity);
-    });
+    #ifdef BE_TEST
+        std::call_once(once_flag, [] {
+            _s_fd_cache = new_lru_cache(config::file_descriptor_cache_capacity);
+        });
+    #else
+        // storage engine may not be opened when doris try to read and write 
+        // temp file under the storage root path. So we need to check it.
+        if (StorageEngine::instance() != nullptr) {
```

Comment:
And also, other place where `_s_fd_cache` is used should also be taken care of. The `release` method can be private, so it will only be called by `close()`

Reply:
Thanks for advices! For problem 1, we can further check weather StorageEngine::instance()->file_cache() == nullptr.